src/Stateless/StateConfiguration.cs

```diff
@@ -415,7 +415,7 @@ public StateConfiguration OnEntry(Action entryAction, string entryActionDescript
                 Enforce.ArgumentNotNull(entryAction, nameof(entryAction));
                 return OnEntry(
                     t => entryAction(),
-                    entryActionDescription ?? entryAction.TryGetMethodName());
+                    entryActionDescription);
```

Comment:
That makes a lot of sense!  I've started using that idea, you can see in my latest commit.